smartsim/_core/_cli/build.py

```diff
@@ -363,8 +366,7 @@ def execute(
 ) -> int:
     verbose = args.v
     keydb = args.keydb
-    device: _TDeviceStr = args.device
-
+    device: Device = Device(args.device.lower())
```

Comment:
Super pedantic style thing: no need to double type `device` here! The pervious type hint was for an implicit cast of `t.Any` -> `_TDeviceStr`,  but mypy should be able to figure out the type of `Device` on its own here!
```suggestion
    device = Device(args.device.lower())
```